esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -206,6 +206,107 @@ WsWUExceptions::WsWUExceptions(IConstWorkUnit& wu): numerr(0), numwrn(0), numinf
+                                         LogAccessReturnColsMode retColsMode, LogAccessLogFormat logFormat, unsigned wuLogSearchTimeBuffSecs)
+{
+    if (!m_remoteLogAccessor)
+        throw makeStringException(ECLWATCH_LOGACCESS_UNAVAILABLE, "WsWuInfo: Remote Log Access plug-in not available!");
+
+    LogAccessConditions logFetchOptions;
+    logFetchOptions.setFilter(getJobIDLogAccessFilter(wuid.str()));
+
+    struct LogAccessTimeRange range;
+    stat_type timeStamp;
+    if (cw->getStatistic(timeStamp, "", StWhenCreated))
+    {
+        CDateTime startt;
+        startt.set(timeStamp / microSecsToSecDivisor);
+
+        startt.adjustTimeSecs(-wuLogSearchTimeBuffSecs);
+        range.setStart(startt);
+
+        StringBuffer newstart;
+        startt.getString(newstart);
+        DBGLOG("Searching for WUID '%s' log entries starting time: '%s'", wuid.str(), newstart.str());
+    }
+    else
+        throw makeStringExceptionV(ECLWATCH_WU_START_NOT_AVAILABLE, "WsWuInfo: Cound not determine WUID '%s' start time", wuid.str());
+
+    if (cw->getStatistic(timeStamp, "", StWhenFinished))
+    {
+        CDateTime endt;
+        endt.set(timeStamp / microSecsToSecDivisor);
+        endt.adjustTimeSecs(wuLogSearchTimeBuffSecs);
+
+        range.setEnd(endt);
+        StringBuffer newend;
+        endt.getString(newend);
+        DBGLOG("Searching for WUID '%s' log entries ending time: '%s'", wuid.str(), newend.str()); //end + time buffer
+    }
+    else
+        WARNLOG("WsWuInfo: Fetching log entries for '%s' without a 'finished'", wuid.str());
+
+    logFetchOptions.setReturnColsMode(retColsMode);
+    logFetchOptions.setTimeRange(range);
+    logFetchOptions.setLimit(maxLogRecords);
+
+    Owned<IFileIOStream> outIOS;
+    if (!isEmptyString(outFile))
+    {
+        CWsWuFileHelper helper(nullptr);
+        outIOS.setown(helper.createIOStreamWithFileName(outFile, IFOcreate));
+    }
+
+    Owned<IRemoteLogAccessStream> logreader = m_remoteLogAccessor->getLogReader(logFetchOptions, logFormat);
+
+    StringBuffer logcontent;
+    unsigned totalRecsRead = 0;
+    unsigned recsRead = 0;
+    
+    if (logFormat == LOGACCESS_LOGFORMAT_json)
+        outIOS->write(11, "{\"lines\": [");
+    else if (logFormat == LOGACCESS_LOGFORMAT_xml)
+        outIOS->write(7, "<lines>");
+
+    while (logreader->readLogEntries(logcontent.clear(), recsRead))
+    {
+        if (logFormat == LOGACCESS_LOGFORMAT_json && totalRecsRead > 0 && recsRead > 0)
+        {
+            writeCharToStream(*outIOS, ',');
+        }
+        totalRecsRead += recsRead;
+        outIOS->write(logcontent.length(), logcontent.str());
+    }
+
+    if (totalRecsRead == maxLogRecords) //Warn of possible truncation
+    {
+        VStringBuffer truncationWarnmessage("Log query reached record limit (%u). Log report could be incomplete.", maxLogRecords);
+        WARNLOG("WsWuInfo: %s", truncationWarnmessage.str());
```

Comment:
This information doesn't seem like it is being reported in the correct way.
Better is likely to be a flag that is returned by the function call, and can be returned in the calling service, or processed some other way.
I'm not really convinced it should be a warning - it is more information.
My concern is that it will mess up anything reading the results.  For the moment that is likely to be human (so won't cause problems) or requested as json.
It can be addressed later.

Reply:
I'll take the blame for this suggestion.
I wanted a way to know from the ZAP report that the log files had been truncated.
An alternative would be to capture the fact that it has been truncated and store as a piece of meta data in the summary report file within the ZAP report.

Reply:
@ghalliday I want to explore how to report this message to log.
I can see why there's hesitancy reporting it as a warn- If the truncation detection was deterministic, it would definitely be a warning (at least from my point of view). But ultimately It's an event which should be investigated and the false positive is not likely to happen often. Reporting it as info has two issues - it's more likely to be filtered out (perhaps ok), and it implies no further action should be taken.

Reply:
Regarding the proper approach to handle possible truncation, I'm not crazy about the in-line message but it gets the message across.
Clearly, bubbling up an indication of truncation is the cleanest approach which the UI would process appropriately.
This functionality is a little odd in that we're creating various files and packaging them into a zip file for download.
Perhaps there could be a "status" file within the zip file which reports any issues encountered? Anyway, it seems a bit outside the scope of this particular PR. if nothing else, I can return the number of records fetched and streamed to local file for future processing.
thoughts @ghalliday @jakesmith

Reply:
The reason for my comments are that this is a general purpose function that happens to currently be called from the zap report code.  If it was called from another context (e.g. eclwatch to implement a paged filtered log view)  then it isn't really a warning to return to the user.  It is expected behaviour.  Similarly the caller would want to know that there might be more.